### PR TITLE
feat: improve screen reader support and general accessibility

### PR DESCRIPTION
**Pull request by United600**
### Settings page
- Remove corner radius fix huynhsontung/Screenbox@60f36e8d250655a96e4b4e865f9ee063056f6d4d, no longer needed CommunityToolkit/Windows#329
- Improve narrator support
- Improve `Audio Visual` expander gamepad navigation
- Add `Audio Visual` icon
- Changed the design of the manage indexer settings card item
- Add related links footer style
- Add third-party license notice link

### Main page
- Add keyboard shortcuts to navigation view items

### Playlist page
- Improve selection command bar
- Add dismiss shortcut to command bar
- Improve narrator support
- Add select all shortcut

### Player
- Improve narrator support
- Adjust some paddings and sizes
- Adjust `Audio Visual` selector position and design

### CODE REVIEW

File: Screenbox/Controls/SeekBar.xaml.cs [hunk 1]
```diff
@@ -46,7 +46,7 @@ public SeekBar()
             RegisterSeekBarPointerHandlers();
             DispatcherQueue dispatcherQueue = DispatcherQueue.GetForCurrentThread();
             _previewToolTipTimer = dispatcherQueue.CreateTimer();
-            _previewToolTip = new ToolTip { Padding = new Thickness(8, 3, 8, 5), FontSize = 15, VerticalOffset = 13 };
+            _previewToolTip = new ToolTip { Padding = new Thickness(8, 3, 8, 5), FontSize = 15, VerticalOffset = 8 };
```
Comment (huynhsontung) [inline Screenbox/Controls/SeekBar.xaml.cs]: Why did you change the vertical offset here to 8? The offset was set to 13 to match the default slider tooltip.

https://github.com/user-attachments/assets/d4f97a9b-e0fd-4170-a96c-5c872d0f6868

File: Screenbox/Pages/PlayerPage.xaml [hunk 2]
```diff
@@ -265,13 +282,46 @@
...
+                x:Name="LivelyOptionsButton"
+                Grid.Column="4"
+                Width="44"
+                Height="44"
+                Margin="4,6,4,2"
+                Padding="0"
+                VerticalAlignment="Top"
+                AutomationProperties.Name="{x:Bind LivelyOptionsButton.(ToolTipService.ToolTip)}"
+                Style="{StaticResource SubtleButtonStyle}"
+                ToolTipService.ToolTip="{strings:Resources Key=VisualizerOptionsToolTip}">
```
Comment (huynhsontung) [inline Screenbox/Pages/PlayerPage.xaml:303]: Now in narrow window sizes, the buttons take up too much real estate. Should we consider resizing the buttons to the standard button size for this layout?

ApplicationFrameHost_dP7QjUhLU5

File: Screenbox/Pages/PlayerPage.xaml [hunk 3]
```diff
@@ -471,24 +504,26 @@
             x:Name="StatusMessage"
             Grid.Row="1"
             Grid.Column="0"
-            Margin="22,4"
-            Padding="11,8,11,9"
-            HorizontalAlignment="Left"
+            Margin="24,4"
+            Padding="11,7,11,8"
+            HorizontalAlignment="Center"
```
Comment (huynhsontung) [inline Screenbox/Pages/PlayerPage.xaml]: Why did you change the horizontal alignment to center? The status message was left aligned so the position is always anchored when the window changes size horizontally.

File: Screenbox/Pages/PlayerPage.xaml [hunk 4]
```diff
@@ -745,41 +781,47 @@
 
             <VisualStateGroup x:Name="LayoutGroup" CurrentStateChanged="OnLayoutVisualStateChanged">
                 <VisualState x:Name="Normal" />
+
                 <VisualState x:Name="Fullscreen">
                     <VisualState.Setters>
                         <Setter Target="HeaderBackground.Height" Value="NaN" />
+                        <Setter Target="TitleBarElement.(Grid.Column)" Value="2" />
+                        <Setter Target="TitleBarElement.(Grid.ColumnSpan)" Value="4" />
```
Comment (huynhsontung) [inline Screenbox/Pages/PlayerPage.xaml]: Why do we need this Grid.Column change in the fullscreen state? I don't think it actually do anything.

File: Screenbox/Pages/PlayerPage.xaml [hunk 5]
```diff
@@ -745,41 +781,47 @@
...
                     </Storyboard>
                 </VisualState>
+
                 <VisualState x:Name="CompactOverlay">
                     <VisualState.Setters>
                         <Setter Target="HeaderBackground.Height" Value="NaN" />
-                        <Setter Target="TitleBarElement.Height" Value="32" />
-                        <Setter Target="TitleBarElement.Margin" Value="0" />
+                        <Setter Target="TitleBarElement.(Grid.Column)" Value="3" />
+                        <Setter Target="TitleBarElement.(Grid.ColumnSpan)" Value="3" />
```
Comment (huynhsontung) [inline Screenbox/Pages/PlayerPage.xaml]: Same grid change here as well. I don't think they really do anything.

File: Screenbox/Pages/PlayerPage.xaml [hunk 6]
```diff
@@ -290,7 +340,7 @@
                 </Grid.RenderTransform>
                 <Image
                     x:Name="BackgroundImageNext"
-                    Source="{x:Bind AlbumArtImage.Source, Mode=OneWay}"
+                    Source="{x:Bind AlbumArtImage.ImageSource, Mode=OneWay}"
```
Comment (huynhsontung) [inline Screenbox/Pages/PlayerPage.xaml:342]: This broke the background fade animation btw. You didn't change [this line](https://github.com/United600/Screenbox/blob/f0440542959b2fd0e9d5a406d2074d5ab5f66c58/Screenbox/Pages/PlayerPage.xaml.cs#L69). I can fix it if you want.

https://github.com/user-attachments/assets/e1ab35e8-c2a2-432e-ad40-28b28ccc77fc

File: Screenbox/Pages/SettingsPage.xaml [hunk 7]
```diff
@@ -17,47 +18,63 @@
...
+    d:DesignHeight="1472"
+    d:DesignWidth="848"
     mc:Ignorable="d">
 
     <Page.Resources>
         <ResourceDictionary>
-            <x:Double x:Key="ToggleSwitchPreContentMargin">4</x:Double>
-            <x:Double x:Key="ToggleSwitchPostContentMargin">4</x:Double>
+            <ResourceDictionary.MergedDictionaries>
+                <ResourceDictionary Source="ms-appx:///CommunityToolkit.WinUI.Controls.SettingsControls/SettingsCard/SettingsCard.xaml" />
```
Comment (huynhsontung) [inline Screenbox/Pages/SettingsPage.xaml:28]: Why do we need to manually import the dictionary here?

File: Screenbox/Strings/en-US/Resources.resw [hunk 8]
```diff
@@ -803,20 +824,26 @@
...
   <data name="SettingsAudioVisualDescription" xml:space="preserve">
-    <value>Choose an imagery style to be displayed on audio playback</value>
+    <value>Choose a backdrop to give your audio playback a visual flair in the immersive view</value>
   </data>
   <data name="PoweredByLivelyWallpaper" xml:space="preserve">
     <value>Powered by Lively Wallpaper</value>
   </data>
   <data name="SettingsImportVisualsHeader" xml:space="preserve">
-    <value>Import visuals</value>
+    <value>Import a visual</value>
```
Comment (huynhsontung) [inline Screenbox/Strings/en-US/Resources.resw]: The button allows you to select multiple files

File: Screenbox/Strings/en-US/Resources.resw [hunk 9]
```diff
@@ -803,20 +824,26 @@
...
   </data>
+  <data name="VisualizerOptionsToolTip" xml:space="preserve">
+    <value>Audio visual options</value>
+  </data>
+  <data name="Visualizers" xml:space="preserve">
+    <value>Visuals</value>
+  </data>
   <data name="BrowseFiles" xml:space="preserve">
-    <value>Browse file(s)</value>
+    <value>Browse files</value>
```
Comment (huynhsontung) [inline Screenbox/Strings/en-US/Resources.resw:837]: Why change this? This style is consistent with the inbox media player.

File: Screenbox/Strings/en-US/Resources.resw [hunk 10]
```diff
@@ -803,20 +824,26 @@
   <data name="GetLivelyVisuals" xml:space="preserve">
     <value>Download visuals on GitHub</value>
   </data>
+  <data name="VisualizerOptionsToolTip" xml:space="preserve">
+    <value>Audio visual options</value>
+  </data>
+  <data name="Visualizers" xml:space="preserve">
+    <value>Visuals</value>
+  </data>
```
Comment (huynhsontung) [inline Screenbox/Strings/en-US/Resources.resw]: Don't create a string called "Visualizers" and then set it to "Visuals". You should either be more descriptive with the string name or match the value with the string name. We may need to use the word "visualizers" in the future.

File: Screenbox/Pages/PlayerPage.xaml (hunk 2 above)
Comment (huynhsontung) [reply Screenbox/Pages/PlayerPage.xaml:303]: Not really an issue but we probably should consider removing the title bar right padding when in fullscreen as well, now that we have a button there.

msedge_abSWyTwdi7

**Review by huynhsontung — CHANGES_REQUESTED** [review]

File: Screenbox/Controls/SeekBar.xaml.cs (hunk 1 above)
Comment (United600) [reply Screenbox/Controls/SeekBar.xaml.cs]: Sorry I'll revert it

File: Screenbox/Pages/PlayerPage.xaml (hunk 3 above)
Comment (United600) [reply Screenbox/Pages/PlayerPage.xaml]: > Why did you change the horizontal alignment to center? The status message was left aligned so the position is always anchored when the window changes size horizontally.

It was to mirror the YouTube player

File: Screenbox/Pages/PlayerPage.xaml (hunk 2 above)
Comment (United600) [reply Screenbox/Pages/PlayerPage.xaml:303]: > Now in narrow window sizes, the buttons take up too much real estate. Should we consider resizing the buttons to the standard button size for this layout?

Personally I think we should slightly reduce it in all layouts, or consider alternatives

> Not really an issue but we probably should consider removing the title bar right padding when in fullscreen as well, now that we have a button there.

AFAIK we can't since it's reserved

imagem

File: Screenbox/Pages/PlayerPage.xaml (hunk 4 above)
Comment (United600) [reply Screenbox/Pages/PlayerPage.xaml]: I was following the LTR issue and, just in case, decided to "correct" the columns, for this specific one it is:

|Padding|Back Button|Queue Button|Title Text|Lively Button|Padding|
|-|-|-|-|-|-|
|Reserved|Collapsed|Visible|Visible|Visible|Reserved|

File: Screenbox/Pages/PlayerPage.xaml (hunk 5 above)
Comment (United600) [reply Screenbox/Pages/PlayerPage.xaml]: And for the CompactOverlay visual state:

|Padding|Back Button|Queue Button|Title Text|Lively Button|Padding|
|-|-|-|-|-|-|
|Reserved|Collapsed|Collapsed|Visible|Collapsed|Reserved|

File: Screenbox/Pages/PlayerPage.xaml (hunk 6 above)
Comment (United600) [reply Screenbox/Pages/PlayerPage.xaml:342]: > This broke the background fade animation btw. You didn't change [this line](https://github.com/United600/Screenbox/blob/f0440542959b2fd0e9d5a406d2074d5ab5f66c58/Screenbox/Pages/PlayerPage.xaml.cs#L69). I can fix it if you want.

Oops, I can fix it.

File: Screenbox/Pages/SettingsPage.xaml (hunk 7 above)
Comment (United600) [reply Screenbox/Pages/SettingsPage.xaml:28]: It's needed for the SettingsCard style

File: Screenbox/Strings/en-US/Resources.resw (hunk 8 above)
Comment (United600) [reply Screenbox/Strings/en-US/Resources.resw]: Windows 11 background settings snippet

I took inspiration from the System Settings

File: Screenbox/Strings/en-US/Resources.resw (hunk 9 above)
Comment (United600) [reply Screenbox/Strings/en-US/Resources.resw:837]: Same as above, also it's better for screen readers and consistent with the System Settings. BTW, is the phrase "Browse file" grammatically correct or logical?

File: Screenbox/Strings/en-US/Resources.resw (hunk 10 above)
Comment (United600) [reply Screenbox/Strings/en-US/Resources.resw]: Oops, it slipped through, I'll fix it

File: Screenbox/Strings/en-US/Resources.resw (hunk 9 above)
Comment (huynhsontung) [reply Screenbox/Strings/en-US/Resources.resw:837]: I would use "Browse files"

File: Screenbox/Strings/en-US/Resources.resw (hunk 8 above)
Comment (huynhsontung) [reply Screenbox/Strings/en-US/Resources.resw]: I see. I have no problem with this then.

File: Screenbox/Pages/PlayerPage.xaml (hunk 4 above)
Comment (huynhsontung) [reply Screenbox/Pages/PlayerPage.xaml]: I think I already fixed the issue in #456. Besides, `TitleBarElement` is just an empty placeholder for a draggable area. There is no content in it. The draggable area should span the whole length of the window, regardless of reserved padding.

File: Screenbox/Pages/SettingsPage.xaml (hunk 7 above)
Comment (huynhsontung) [reply Screenbox/Pages/SettingsPage.xaml:28]: The SettingsCard style is not imported automatically?

File: Screenbox/Pages/PlayerPage.xaml (hunk 4 above)
Comment (United600) [reply Screenbox/Pages/PlayerPage.xaml]: You're right and this `TitleBar` is exceptional

File: Screenbox/Pages/SettingsPage.xaml (hunk 7 above)
Comment (United600) [reply Screenbox/Pages/SettingsPage.xaml:28]: You'd think so, but it won't build without it

File: Screenbox/Strings/en-US/Resources.resw (hunk 8 above)
Comment (United600) [reply Screenbox/Strings/en-US/Resources.resw]: Already reverted, its fine either way. If you want I can re-revert it?

File: Screenbox/Pages/PlayerPage.xaml [hunk 11]
```diff
@@ -498,10 +531,8 @@
             Grid.Row="0"
             Grid.RowSpan="3"
             Grid.Column="0"
-            Width="72"
-            Height="72"
-            HorizontalAlignment="Center"
-            VerticalAlignment="Center"
+            Width="96"
+            Height="96"
```
Comment (huynhsontung) [inline Screenbox/Pages/PlayerPage.xaml]: The badge looks huge??

Comparing to YouTube player of the same size

ApplicationFrameHost_2DqcCwOfrJ
msedge_GvxOrBRAeh

File: Screenbox/Pages/PlayerPage.xaml [hunk 12]
```diff
@@ -512,14 +543,14 @@
...
+                <animations:ScaleAnimation To="1.25" Duration="0:0:0.6" />
             </animations:Implicit.HideAnimations>
             <FontIcon
                 x:Name="PlayerStateBadgeIcon"
-                HorizontalAlignment="Center"
-                VerticalAlignment="Center"
+                d:Glyph="&#xF5B0;"
                 FontFamily="{StaticResource ScreenboxSymbolThemeFontFamily}"
-                FontSize="{StaticResource TitleTextBlockFontSize}"
+                FontSize="32"
```
Comment (huynhsontung) [inline Screenbox/Pages/PlayerPage.xaml]: From 24 to 32?

File: Screenbox/Pages/PlayerPage.xaml (hunk 11 above)
Comment (United600) [reply Screenbox/Pages/PlayerPage.xaml]: Increased on the normal state and decreased on the compact overlay state, making for a fairer comparison in my opinion
